docs/design_proposals/sync-improvements.md

```diff
@@ -0,0 +1,319 @@
+# Sync improvements
+
+* Author(s): Cornelius Weig (@corneliusweig)
+* Design Shepherd: Tejal Desai (@tejal29)
+* Date: 03/20/2019
+* Status: Draft
+
+## Background
+
+Currently, skaffold config supports `artifact.sync` as a way to sync files
+directly to pods. So far, artifact sync requires a specification of sync
+patterns like
+
+```yaml
+sync:
+  '*.js': app/
+```
+
+This is error prone and unnecessarily hard to use, because the destination is
```

Comment:
```suggestion
This is error prone and unnecessarily hard to use because the destination can often be determined by the builder. For example a destination path is
```